ui/app/controllers/clients/index.js

```diff
@@ -72,16 +77,31 @@ export default Controller.extend(
       return datacenters.sort().map(dc => ({ key: dc, label: dc }));
     }),
 
+    optionsVolume: computed('nodes.[]', function() {
+      const flatten = (acc, val) => acc.concat(val.toArray());
+
+      const allVolumes = this.nodes.mapBy('hostVolumes').reduce(flatten, []);
+      const volumes = Array.from(new Set(allVolumes.mapBy('name')));
+
+      scheduleOnce('actions', () => {
+        this.set('qpVolume', serialize(intersection(volumes, this.selectionVolume)));
+      });
```

Comment:
I see that this is a pattern that already exists above in `optionsDatacenter` but I’m a bit surprised to see mutation in a computed property like this. I wonder whether we’ll be able to move away from having to do it eventually?